extensions/sql-database-projects/src/models/project.ts

```diff
@@ -158,15 +158,39 @@ export class Project implements ISqlProject {
 
 		await proj.readProjFile();
 
-		if (!proj.isCrossPlatformCompatible && promptIfNeedsUpdating) {
-			const result = await window.showWarningMessage(constants.updateProjectForRoundTrip(proj.projectFileName), constants.yesString, constants.noString);
+		if (promptIfNeedsUpdating) {
+			await this.checkPromptCrossPlatStatus(proj, false /* don't block the thread until the  prompt*/);
+		}
+
+		return proj;
+	}
 
+	/**
+	 * If project does not support cross-plat building, prompts the user for whether to update and updates if accepted
+	 * @param project
+	 * @param blockingPrompt whether to block the thread until the user updates, or to fire and forget
+	 * @returns true if the project is updated after return, false if the user rejected the prompt
+	 */
+	public static async checkPromptCrossPlatStatus(project: Project, blockingPrompt: boolean): Promise<boolean> {
+		if (project.isCrossPlatformCompatible) {
+			return true;
+		}
+
+		if (blockingPrompt) {
```

Comment:
if it's a blocking prompt, should it be a modal dialog so people don't miss it?

Reply:
Yeah, good call

Reply:
I'm going to check in with the modal, but I figured I'd show what both look like in case we decide to change it later.